crates/oxc_linter/src/rules/eslint/no_unused_vars/symbol.rs

```diff
@@ -0,0 +1,225 @@
+    }
+
+    #[inline]
+    pub fn id(&self) -> SymbolId {
+        self.id
+    }
+
+    #[inline]
+    pub fn name(&self) -> &str {
+        self.symbols().get_name(self.id)
+    }
+
+    /// [`Span`] for the node declaring the [`Symbol`].
+    #[inline]
+    pub fn span(&self) -> Span {
+        self.symbols().get_span(self.id)
+    }
+
+    #[inline]
+    pub const fn flags(&self) -> SymbolFlags {
+        self.flags
+    }
+
+    #[inline]
+    pub fn scope_id(&self) -> ScopeId {
+        self.symbols().get_scope_id(self.id)
+    }
+
+    #[inline]
+    pub fn declaration(&self) -> &AstNode<'a> {
+        self.nodes().get_node(self.declaration_id())
+    }
+
+    #[inline]
+    pub fn references(&self) -> impl DoubleEndedIterator<Item = &Reference> + '_ {
+        self.symbols().get_resolved_references(self.id)
+    }
+
+    /// Is this [`Symbol`] declared in the root scope?
+    pub fn is_root(&self) -> bool {
+        self.symbols().get_scope_id(self.id) == self.scopes().root_scope_id()
+    }
+
+    #[inline]
+    fn declaration_id(&self) -> AstNodeId {
+        self.symbols().get_declaration(self.id)
+    }
+
+    #[inline]
+    pub fn nodes(&self) -> &AstNodes<'a> {
+        self.semantic.nodes()
+    }
+
+    #[inline]
+    pub fn scopes(&self) -> &ScopeTree {
+        self.semantic.scopes()
+    }
+
+    #[inline]
+    pub fn symbols(&self) -> &SymbolTable {
+        self.semantic.symbols()
+    }
+
+    pub fn iter_parents(&self) -> impl Iterator<Item = &AstNode<'a>> + '_ {
+        self.nodes().iter_parents(self.declaration_id()).skip(1)
+    }
+
+    pub fn iter_relevant_parents(
+        &self,
+        node_id: AstNodeId,
+    ) -> impl Iterator<Item = &AstNode<'a>> + Clone + '_ {
+        self.nodes().iter_parents(node_id).skip(1).filter(|n| Self::is_relevant_kind(n.kind()))
+    }
+
+    pub fn iter_relevant_parent_and_grandparent_kinds(
+        &self,
+        node_id: AstNodeId,
+    ) -> impl Iterator<Item = (/* parent */ AstKind<'a>, /* grandparent */ AstKind<'a>)> + Clone + '_
+    {
+        let parents_iter = self
+            .nodes()
+            .iter_parents(node_id)
+            .map(AstNode::kind)
+            // no skip
+            .filter(|kind| Self::is_relevant_kind(*kind));
+
+        let grandparents_iter = parents_iter.clone().skip(1);
+
+        parents_iter.zip(grandparents_iter)
+    }
+
+    #[inline]
+    const fn is_relevant_kind(kind: AstKind<'a>) -> bool {
+        !matches!(kind, AstKind::ParenthesizedExpression(_))
+    }
+}
+
+impl<'s, 'a> Symbol<'s, 'a> {
+    /// Is this [`Symbol`] exported?
+    ///
+    /// NOTE: does not support CJS right now.
+    pub fn is_exported(&self) -> bool {
+        let is_in_exportable_scope = self.is_root() || self.is_in_ts_namespace();
+        (is_in_exportable_scope
+            && (self.flags.contains(SymbolFlags::Export)
+                || self.semantic.module_record().exported_bindings.contains_key(self.name())))
+            || self.in_export_node()
```

Comment:
It looks like we can simplify to `self.flags.contains(SymbolFlags::Export) || (self.is_in_ts_namespace() && self.in_export_node())` ?

Reply:
Potentially, depending on how well symbol flags are working now. That was the main blocker.

Reply:
The module record check is needed for scenarios like this:
```ts
const foo = 1
export { foo }
```